vertical-pod-autoscaler/pkg/recommender/input/metrics/metrics_datadog_client.go

```diff
@@ -217,26 +267,74 @@ func (d ddclientPodMetrics) List(_ context.Context, _ metav1.ListOptions) (*v1be
 
 	podItems := make([]v1beta1.PodMetrics, 0, len(podCpus))
 	for podname, cpuVals := range podCpus {
-		podItems = append(podItems, *d.aggregatePodMetrics(cpuVals, podMems[podname]))
+		podMets := d.aggregatePodMetrics(podname, cpuVals, podMems[podname])
```

Comment:
I think this code contains a bug: Namespace is always "", so we get all the metrics for all the pods of cluster (does it work at scale is another question?) , and later we index by podname.... what if we have 2 pods with the same name in different namespaces ?
To be investigated/fixed

Reply:
Can you add a TODO in the code ?